src/drivers/magnetometer/isentek/ist8310/IST8310.cpp

```diff
@@ -85,19 +85,27 @@ void IST8310::print_status()
 
 int IST8310::probe()
 {
-	// Apparently, the IST8310's WHOAMI register is writeable. Presumably,
-	// this can get corrupted by bus noise. It is only reset if powered off
-	// for 30s or by a reset.
-	//
-	// Therefore, we do a reset before doing the WHOAMI.
-	RegisterWrite(Register::CNTL2, CNTL2_BIT::SRST);
+	uint8_t id = RegisterRead(Register::WAI);
 
-	px4_usleep(10000);
+	if (id != Device_ID) {
+		DEVICE_DEBUG("unexpected WAI 0x%02x", id);
 
-	const uint8_t WAI = RegisterRead(Register::WAI);
+		// Apparently, the IST8310's WHOAMI register is writeable. Presumably,
+		// this can get corrupted by bus noise. It is only reset if powered off
+		// for 30s or by a reset.
+		RegisterWrite(Register::CNTL2, CNTL2_BIT::SRST);
+
+		auto start_time = hrt_absolute_time();
+
+		while (hrt_elapsed_time(&start_time) < 50_ms) {
+			px4_usleep(10_ms);
```

Comment:
This is nitpicky, but we should really only be using our time literals (that we get from HRT) with HRT. At the moment it's fine, but there's no guarantee the underlying storage is an integer in microseconds (it could be nanoseconds in the future).

```suggestion
			px4_usleep(10'000);
```